src/crab/ebpf_domain.cpp

```diff
@@ -1472,6 +1550,13 @@ void ebpf_domain_t::operator()(const Bin& bin) {
                             inv -= dst_offset.value();
                         }
                         inv -= dst.shared_region_size;
+                        auto lb = m_inv.eval_interval(src.value).lb().number();
+                        if (lb && lb.value().fits_sint() && ((int)lb.value() > 0))
+                            // Narrow the numeric size.
+                            apply(inv, crab::arith_binop_t::SUB, dst.stack_numeric_size, dst.stack_numeric_size, src.value, false);
```

Comment:
Seems incorrect. 
```suggestion
                        auto ub = m_inv.eval_interval(src.value).lb().number();
                        auto lb = m_inv.eval_interval(src.value).lb().number();
                        if (ub && ub.value().fits_sint() && ((int)ub.value() < 0) && lb && lb.value().fits_sint())
                            // If it must be a negative number, narrow the numeric size.
                            apply(inv, crab::arith_binop_t::ADD, dst.stack_numeric_size, dst.stack_numeric_size, (int)lb, false);
```

Reply:
Updated (this location was simplified further since the result is T_NUM) but I did your suggestion in another location in this file where it did matter.